facets_atlasmaker/utils/README.md

```diff
@@ -0,0 +1,14 @@
+# Atlasmaker Utilities
+
+This directory contains utilities that can be useful for testing or  experimenting with Atlasmaker.
+
+## Wikipedia Sourcelist Generator
+
+The script connects to the Wikipedia API to get a list of URLs of Featured Images that 
+you can use as input to Atlasmaker.
+
+Example usage for getting a list of 1000 images with debug messages printed to stdout:
+
+```sh
+bazel run :wikipedia_sourcelist_generator -- --num_images=1000 --outputdir=$PWD --verbosity=1
```

Comment:
The sourcelist generator worked fine for creating a long list, thanks! Was able to make a list of 10k images.

However, when I tried to generate an atlas with these 10k images, it seemed to be doing OK for 15 minutes and then blew up. I'm not sure why, but here's the top and bottom of the very long stack:

```
[Parallel(n_jobs=-1)]: Done 2576 tasks      | elapsed: 14.6min
E0629 14:57:10.262512 140735762211712 parallelize.py:33] Retrieval of file https://upload.wikimedia.org/wikipedia/commons/2/2a/Coat_of_arms_of_Mexico.svg failed with error: cannot identify image file <_io.BytesIO object at 0x109333d10>
E0629 14:57:11.501877 140735762211712 parallelize.py:33] Retrieval of file https://upload.wikimedia.org/wikipedia/commons/d/d4/Coat_of_Arms_of_Siam_%281873-1910%29.svg failed with error: cannot identify image file <_io.BytesIO object at 0x1093286b0>
E0629 14:57:12.611440 140735762211712 parallelize.py:33] Retrieval of file https://upload.wikimedia.org/wikipedia/commons/a/a1/Coat_of_arms_of_Trinidad_and_Tobago.svg failed with error: cannot identify image file <_io.BytesIO object at 0x109333e30>
E0629 14:57:15.713234 140735762211712 parallelize.py:33] Retrieval of file https://upload.wikimedia.org/wikipedia/commons/d/d6/College_of_Arms-Lant%27s_Roll.svg failed with error: cannot identify image file <_io.BytesIO object at 0x116457710>
E0629 14:57:39.827016 140735762211712 parallelize.py:33] Retrieval of file https://upload.wikimedia.org/wikipedia/commons/6/62/Cheetahs_on_the_Edge_%28Director%27s_Cut%29.ogv failed with error: cannot identify image file <_io.BytesIO object at 0x1166dec50>
E0629 14:59:09.965570 140735762211712 parallelize.py:33] Retrieval of file https://upload.wikimedia.org/wikipedia/commons/c/c0/Big_Buck_Bunny_4K.webm failed with error: cannot identify image file <_io.BytesIO object at 0x109335b30>

Joblib's exception reporting continues...

Traceback (most recent call last):
  File "/private/var/tmp/_bazel_jimbo/bdc924cd5f9f9e5884088ec4eae7b2c1/execroot/ai_google_pair_facets/bazel-out/darwin-fastbuild/bin/facets_atlasmaker/atlasmaker.runfiles/ai_google_pair_facets/facets_atlasmaker/atlasmaker.py", line 182, in <module>
    app.run(main)
  File "/Users/jimbo/facets-env/lib/python2.7/site-packages/absl/app.py", line 274, in run
    _run_main(main, args)
  File "/Users/jimbo/facets-env/lib/python2.7/site-packages/absl/app.py", line 238, in _run_main
    sys.exit(main(argv))
  File "/private/var/tmp/_bazel_jimbo/bdc924cd5f9f9e5884088ec4eae7b2c1/execroot/ai_google_pair_facets/bazel-out/darwin-fastbuild/bin/facets_atlasmaker/atlasmaker.runfiles/ai_google_pair_facets/facets_atlasmaker/atlasmaker.py", line 164, in main
    n_jobs=FLAGS.num_parallel_jobs, verbose=FLAGS.parallelization_verbosity)
  File "/Users/jimbo/checkouts/facets/facets_atlasmaker/parallelize.py", line 70, in get_and_convert_images_parallel
    for location in image_src_locations)
  File "/Users/jimbo/facets-env/lib/python2.7/site-packages/joblib/parallel.py", line 789, in __call__
    self.retrieve()
  File "/Users/jimbo/facets-env/lib/python2.7/site-packages/joblib/parallel.py", line 740, in retrieve
    raise exception
joblib.my_exceptions.JoblibIOError: JoblibIOError
___________________________________________________________________________
Multiprocessing exception:
```
...
```
...........................................................................
/Users/jimbo/facets-env/lib/python2.7/site-packages/PIL/Image.py in resize(self=<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=648x432>, size=(80, 53), resample=1, box=(0, 0, 648, 432))
   1742             return self.convert('La').resize(size, resample, box).convert('LA')
   1743 
   1744         if self.mode == 'RGBA':
   1745             return self.convert('RGBa').resize(size, resample, box).convert('RGBA')
   1746 
-> 1747         self.load()
        self.load = <bound method JpegImageFile.load of <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=648x432>>
   1748 
   1749         return self._new(self.im.resize(size, resample, box))
   1750 
   1751     def rotate(self, angle, resample=NEAREST, expand=0, center=None,

...........................................................................
/Users/jimbo/facets-env/lib/python2.7/site-packages/PIL/ImageFile.py in load(self=<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=648x432>)
    223                                 if LOAD_TRUNCATED_IMAGES:
    224                                     break
    225                                 else:
    226                                     self.tile = []
    227                                     raise IOError("image file is truncated "
--> 228                                                   "(%d bytes not processed)" % len(b))
        b = '\x95^\xb0V\xb6\x15\xb9EC\xd6'
    229 
    230                             b = b + s
    231                             n, err_code = decoder.decode(b)
    232                             if n < 0:

IOError: image file is truncated (10 bytes not processed)
___________________________________________________________________________
```

You don't have to fix this, just pasting in case it's something obvious to you. I'm running atlasmaker again with 1,000 images.

Reply:
1,000 image atlasmaker worked great! Here's the end of the log:

```
[Parallel(n_jobs=-1)]: Done 952 tasks      | elapsed:  5.6min
[Parallel(n_jobs=-1)]: Done 1000 out of 1000 | elapsed:  5.8min finished
I0629 15:12:59.930185 140735762211712 montage.py:175] Montaged 1000 images onto sprite atlas of size (2560, 2560) pixels.
W0629 15:12:59.930639 140735762211712 montage.py:178] 12 images had failures and were replaced by the default image
```

I liked that it told me how many images failed. Here's the top of the `manifest.json` file:

```json
{
  "source image": "https://upload.wikimedia.org/wikipedia/commons/e/e4/1_taipei_sunrise_panorama_dxr_edit_pangen_141215_1.jpg",
  "offset_x": 0,
  "image_name": "1_taipei_sunrise_panorama_dxr_edit_pangen_141215_1.jpg",
  "success": true,
  "offset_y": 0
}
```

Reply:
Error message is currently the entire message, which is much more descriptive. I can truncate this to a more concise message if needed, but for now it may be more helpful to have the full error message.